docs/topics/gradle/gradle-compilation-and-caches.md

```diff
@@ -0,0 +1,340 @@
+  In this case, you should provide a different set of JVM arguments for such a module, so a Kotlin daemon with a larger heap size would start only for developers who touch this specific module.
+  > If you are already running a Kotlin daemon that has enough heap size to handle the compilation request,
+  > even if other requested JVM arguments are different, this daemon will be reused instead of starting a new one.
+  >
+  {type="note"}
+* If the `Xmx` is not specified, the Kotlin daemon will inherit it from the Gradle daemon.
+
+## Defining Kotlin compiler execution strategy
+
+_Kotlin compiler execution strategy_ defines where the Kotlin compiler is executed and if incremental compilation is supported in each case.
+
+There are three compiler execution strategies:
+
+| Strategy       | Where Kotlin compiler is executed          | Incremental compilation | Other characteristics and notes                                                                                                                                                                                                                                                |
+|----------------|--------------------------------------------|-------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Daemon         | Inside its own daemon process              | Yes                     | *The default and the fastest strategy*. Can be shared between different Gradle daemons and multiple parallel compilations.                                                                                                                                                     |
+| In process     | Inside the Gradle daemon process           | No                      | May share the heap with the Gradle daemon. The "In process" execution strategy is _slower_ than the "Daemon" execution strategy. Each [worker](https://docs.gradle.org/current/userguide/worker_api.html) creates a separate Kotlin compiler classloader for each compilation. |
+| Out of process | In a separate process for each compilation | No                      | The slowest execution strategy. Similar to the "In process", but additionally creates a separate Java process within a Gradle worker for each compilation.                                                                                                                     |
+
+To define a Kotlin compiler execution strategy, you can use one of the following properties:
+* The `kotlin.compiler.execution.strategy` Gradle property.
+* The `compilerExecutionStrategy` compile task property.
+* The deprecated `-Dkotlin.compiler.execution.strategy` system property, which will be removed in future releases.
```

Comment:
Should we mention it if we're going to deprecate it soon?

Reply:
Yes, we should because it's still an option.